From 16bb4a036b87e92d1bf186f5dfe735463c85434d Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Wed, 12 Jun 2024 15:36:37 +0800
Subject: [PATCH 128/133] loongarch(loongarch.cc): drop broken op_mode
 optimisation

This can technically shorten the branch but does not interoperate with the
current vectorisation macros.

Signed-off-by: Mingcong Bai <baimingcong@uniontech.com>
---
 gcc/config/loongarch/loongarch.cc | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/gcc/config/loongarch/loongarch.cc b/gcc/config/loongarch/loongarch.cc
index d30d5dd7468..d0db08c8270 100644
--- a/src/gcc/config/loongarch/loongarch.cc
+++ b/src/gcc/config/loongarch/loongarch.cc
@@ -9625,13 +9625,10 @@ expand_perm_const_2_end:
 /* Implement TARGET_VECTORIZE_VEC_PERM_CONST.  */
 
 static bool
-loongarch_vectorize_vec_perm_const (machine_mode vmode, machine_mode op_mode,
+loongarch_vectorize_vec_perm_const (machine_mode vmode,
                    rtx target, rtx op0, rtx op1,
                    const vec_perm_indices &sel)
 {
-  if (vmode != op_mode)
-    return false;
-
   struct expand_vec_perm_d d;
   int i, nelt, which;
   unsigned char orig_perm[MAX_VECT_LEN];
-- 
2.43.4

